osu.Game/Overlays/SocialOverlay.cs

```diff
@@ -166,20 +171,51 @@ private void recreatePanels(PanelDisplayStyle displayStyle)
 
             LoadComponentAsync(newPanels, f =>
             {
-                if (panels != null)
-                    ScrollFlow.Remove(panels);
-
+                loading.Hide();
                 ScrollFlow.Add(panels = newPanels);
             });
         }
 
         private void updateUsers(IEnumerable<User> newUsers)
         {
-            Users = newUsers;
-            loading.Hide();
+            loading.Show();
+
+            var sortDirection = Filter.DisplayStyleControl.Dropdown.Current.Value;
+
+            IEnumerable<User> sortedUsers = newUsers;
+
+            if (sortedUsers.Any())
+            {
+                switch (Filter.Tabs.Current.Value)
+                {
+                    case SocialSortCriteria.Location:
+                        sortedUsers = sortBy(sortedUsers, u => u.Country.FullName, sortDirection);
+                        break;
+
+                    case SocialSortCriteria.Name:
+                        sortedUsers = sortBy(sortedUsers, u => u.Username, sortDirection);
+                        break;
+                }
+            }
+
+            Users = sortedUsers;
             recreatePanels(Filter.DisplayStyleControl.DisplayStyle.Value);
         }
 
+        private IEnumerable<User> sortBy<T>(IEnumerable<User> users, Func<User, T> condition, SortDirection sortDirection) =>
+            sortDirection == SortDirection.Ascending ? users.OrderBy(condition) : users.OrderByDescending(condition);
+
+        private void onDropdownChanged()
+        {
+            if (Users == null)
+            {
+                queueUpdate();
```

Comment:
i don't think this needs to exist. either there is an already pending update or there are no users, no?